packaging/flavorgen/flavors/generators.go

```diff
@@ -67,7 +67,7 @@ systemd:
       RemainAfterExit=yes
       Environment=OUTPUT=/run/metadata/coreos
       ExecStart=/usr/bin/mkdir --parent /run/metadata
-      ExecStart=/usr/bin/bash -cv 'echo "COREOS_CUSTOM_HOSTNAME=$(/usr/share/oem/bin/vmtoolsd --cmd "info-get guestinfo.metadata" | base64 -d | grep local-hostname | awk {\'print $2\'} | tr -d \'"\')" > $${OUTPUT}'
+      ExecStart=/usr/bin/bash -cv 'VMTOOLSD_PATH=$(find /usr -iname 'vmtoolsd' -executable); if [[ -n "$vmtoolsd_path" ]]; then echo "COREOS_CUSTOM_HOSTNAME=$("$VMTOOLSD_PATH" --cmd "info-get guestinfo.metadata" | base64 -d | grep local-hostname | awk {\'print $2\'} | tr -d \'"\')" > $${OUTPUT}; else echo "vmtoolsd not found in /usr/share/oem/bin/ or /usr/bin/"; fi'
```

Comment:
Let's add a comment on why we are doing this :-) (at place X for version Y and place a for version B).

What do you think about only using if / else? The path should not change too much right?

```suggestion
      ExecStart=/usr/bin/bash -cv 'echo "COREOS_CUSTOM_HOSTNAME=$("$(find /usr/bin /usr/share/oem -name vmtoolsd -type f -executable 2>/dev/null | head -n 1)" --cmd "info-get guestinfo.metadata" | base64 -d | grep local-hostname | awk {\'print $2\'} | tr -d \'"\')" > $${OUTPUT}'
```

Reply:
Maybe only use find to gather the command and let it fail if it fails?

we need `-type f` because directories can also be executable.

Also I think we don't have to search through the whole `/usr` directory. better to have some well-known paths to search in.

Reply:
WDYT?

Reply:
Thanks for the suggestion!
I will try this one out. 👍

Reply:
Hey @chrischdi ,
i tried your suggestion on flatcar version 3602 and 3760. 
I was able to successfully spawn the control plane and the worker nodes.